tests/application_development/libcxx/testcase.yaml

```diff
@@ -1,8 +1,15 @@
 tests:
   misc.app_dev.libcxx:
+    arch_exclude: posix
+    platform_exclude: qemu_x86_coverage qemu_x86_64 qemu_x86_long
```

Comment:
Why are these excluded?
This all looks really suspicious. Can we do the right thing with a "filter:" expression, or "min_ram:" or whatever it is that prevents the test running on these boards instead?

Reply:
Best guess from https://github.com/zephyrproject-rtos/zephyr/pull/19507#issuecomment-537006768 is this is either a docker issue or a toolchain issue related to multilib x86 toolchain builds.

Reply:
Good one on the `min_ram` I will update accordingly for `96b_meerkat96 colibri_imx7d_m4 warp7_m4 pico_pi_m4` with `min_flash` as it was a ROM size limitation.

For the `qemu_x86_coverage qemu_x86_64` I have no idea and was copied from the initial test case in th yaml file.

The `qemu_x86_long` looks like a toolchain issue.

As @pabigot mentioned the `arch_exclude: posix` is a CI docker image limitation.

Reply:
OK.

Do we have bugs open on the x86 issues?

Reply:
> Do we have bugs open on the x86 issues?

I don't think there is an issue for `qemu_x86_long` toolchain problem.


Further did a quick test  for`qemu_x86_coverage` and `qemu_x86_64`

`qemu_x86_coverage` just hangs

`qemu_x86_64` fails to compile:
```
zephyr-sdk-0.10.3/x86_64-zephyr-elf/x86_64-zephyr-elf/include/c++/8.3.0/limits:1599:35: error: SSE register return with SSE disabled
       min() _GLIBCXX_USE_NOEXCEPT { return __FLT_MIN__; }
                                   ^
```

Reply:
I think #19735 will help with the x86 issues but for some reason it completely fails shippable right now.